pan-domain-auth-core/src/main/scala/com/gu/pandomainauth/service/S3Bucket.scala

```diff
@@ -1,29 +1,52 @@
 package com.gu.pandomainauth.service
 
 import java.util.Properties
+
 import com.amazonaws.ClientConfiguration
-import com.amazonaws.regions.{Regions, Region}
+import com.amazonaws.regions.{Region, Regions}
 import com.gu.pandomainauth.PublicSettings
 
 import scala.collection.JavaConverters._
 import com.amazonaws.auth.AWSCredentialsProvider
 import com.amazonaws.services.s3.AmazonS3Client
 import com.amazonaws.services.s3.model.GetObjectRequest
 
+import scala.concurrent.duration.DurationDouble
+import scala.concurrent.{Await, Future, Promise}
+import scala.util.Try
+
 class S3Bucket(credentialsProvider: AWSCredentialsProvider, regionOption: Option[Region] = None, proxyConfiguration: Option[ProxyConfiguration] = None) {
 
   val region = regionOption getOrElse (Region getRegion Regions.EU_WEST_1)
   val s3Client = region.createClient(classOf[AmazonS3Client], credentialsProvider, awsClientConfiguration)
 
   val bucketName = PublicSettings.bucketName
 
-  def readDomainSettings(domain: String): Map[String, String] = {
+  def readDomainSettingsBlocking(domain: String): Map[String, String] = {
+    Await.result(readDomainSettingsAsync(domain), 30.seconds)
+  }
+
+  def readDomainSettingsAsync(domain: String): Future[Map[String, String]] = {
+    val promise = Promise[Map[String, String]]
+
+    def fetch: Unit = {
+      val result = Try {
+        val domainSecretFile = s3Client.getObject(new GetObjectRequest(bucketName, domain + ".settings"))
+        val props = new Properties()
+
+        props.load(domainSecretFile.getObjectContent)
+        props.asScala.toMap
+      }
+      promise.complete(result)
+    }
 
-    val domainSecretFile = s3Client.getObject(new GetObjectRequest(bucketName, domain + ".settings"))
-    val props = new Properties()
+    // The java way to span a new thread in order to block elsewhere than on Play's actor system
+    val thread = new Thread(() => fetch)
```

Comment:
That's where I would like feedback. It's cheap but really not ideal

Reply:
Can you use this [`TransferManager`](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/index.html?com/amazonaws/services/s3/transfer/TransferManager.html) class? It looks like this is what gives you asynchronous powers with S3 ⚡️

Reply:
Had never heard of that, it looks useful. Unfortunately there's no method to download to a buffer. It looks like it's only writing to a `File` and must be optimised for large file workflows.